scripts/keys.coffee

```diff
@@ -22,81 +22,213 @@
 
 module.exports = (robot)->
 
-	key = ()->
-		Key = robot.brain.get("key") or ""
-		robot.brain.set("key" ,Key)
+	kb_key = ()->
+		Key = robot.brain.get("kb_key") or ""
+		robot.brain.set("kb_key" ,Key)
 		Key	
 
-	
-	robot.respond /i have (a key|the key|key|keys)/i, (msg)->
+	ravindra_key = ()->
+		Key = robot.brain.get("ravindra_key") or ""
+		robot.brain.set("ravindra_key" ,Key)
+		Key
+
+	master_key = ()->
+		Key = robot.brain.get("master_key") or ""
+		robot.brain.set("master_key" ,Key)
+		Key
+
+
+#this section belongs to "i have keys"
+	robot.respond /i have (.+) (a key|the key|key|keys)/i, (msg)->
+		keyname = msg.match[1]
 		name = msg.message.user.name 
 		user = robot.brain.userForName name
-		k = key()
-		if typeof user is 'object'
-			msg.send "Okay #{name} has keys"
-			k = user.name
-		robot.brain.set("key",k)	
-
+		if keyname is 'ravindra' or 'rv'
+			k = ravindra_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
+				k = user.name
+			robot.brain.set("ravindra_key",k)
+		else if keyname is 'kb' or 'kasturba'
+			k = kb_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
+				k = user.name
+			robot.brain.set("kb_key",k)
+		else if keyname is 'master'
+			k = root_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
+				k = user.name
+			robot.brain.set("master_key",k)	
 
-	robot.respond /i (don\'t|dont|do not) (has|have) (the key|key|keys|a key)/i , (msg)->
+#this section belongs to "i dont have key"
+	robot.respond /i (don\'t|dont|do not) (has|have) (.+) (the key|key|keys|a key)/i , (msg)->
+		keyname = msg.match[3]
 		name = msg.message.user.name
 		user = robot.brain.userForName name
-		k = key()
-		if k is name
-			k = ""
-		if typeof user is 'object'
-			if k is ""
-				msg.send "Okay #{name} doesn't have keys. Who got the keys then?"
-			else
-				msg.send "Yes , i know buddy, its because #{k} has got the keys"	
-		robot.brain.set("key",k)	
-
+		
+		if keyname is 'ravindra' || keyname is 'rv'
+			k = ravindra_key()
```

Comment:
Same goes here. Repetition of code means that something can be done even better.